gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DagProc.java

```diff
@@ -35,38 +36,39 @@
 
 /**
  * Responsible for performing the actual work for a given {@link DagTask} by first initializing its state, performing
- * actions based on the type of {@link DagTask} and finally submitting an event to the executor.
+ * actions based on the type of {@link DagTask}. Submitting events in time is important (PR#3641), hence initialize and
+ * act methods submit events as they happen.
  */
 @Alpha
 @Slf4j
 @RequiredArgsConstructor
-public abstract class DagProc<S, T> {
+public abstract class DagProc<S> {
```

Comment:
looks reasonable to eliminate `T` param, although would now be customary to rename `S` to `T`.  please also describe the meaning of the generic type in the javadoc